utils/pattern/src/frontend/mod.rs

```diff
@@ -191,6 +191,31 @@ where
     }
 }
 
+impl<B> Pattern<B, B::Store>
+where
+    B: PatternBackend,
+{
+    /// Creates a pattern from a borrowed store.
+    ///
+    /// # Safety
+    ///
+    /// This is a low-level method that bypasses the store validation and assumes the store
+    /// is valid. It is primarily used by the backend implementations.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu_pattern::Pattern;
+    /// use icu_pattern::SinglePlaceholder;
+    ///
+    /// Pattern::<SinglePlaceholder, _>::from_borrowed_store_unchecked("\x01 days")
+    ///     .expect("valid pattern");
+    /// ```
+    pub unsafe fn from_borrowed_store_unchecked(store: &B::Store) -> &Self {
+        &*(store as *const B::Store as *const Self)
```

Comment:
explain in a comment why this is safe


```

The layout of `Pattern` type  is the same as `Store` because `_backend` field is zero sized and store is the only other field.
```